modelcluster/queryset.py

```diff
@@ -516,6 +517,21 @@ def order_by(self, *fields):
         clone = self.get_clone(results=self.results[:])
         sort_by_fields(clone.results, fields)
         return clone
+    
+    def distinct(self, *fields):
+        if fields and connection.vendor != 'postgresql':
+            raise NotSupportedError("DISTINCT ON fields is not supported by this database backend")
```

Comment:
Is this needed, given that we're bypassing the database here? I guess you're trying to keep FakeQuerySet as close as possible to the real database behaviour, but I think it's fine for FakeQuerySet to have extra functionality that the database doesn't support.